app/services/meta-tags.ts

```diff
@@ -0,0 +1,159 @@
+import { service } from '@ember-decorators/service';
+import Service from '@ember/service';
+import config from 'ember-get-config';
+import pathJoin from 'ember-osf-web/utils/path-join';
+import toArray from 'ember-osf-web/utils/to-array';
+
+export type Content = string | number | null | undefined;
+
+export type DataContent = Content | Content[];
+
+export interface MetaTagsData {
+    title?: DataContent;
+    type?: DataContent;
+    description?: DataContent;
+    url?: DataContent;
+    doi?: DataContent;
+    identifier?: DataContent;
+    publishedDate?: DataContent;
+    modifiedDate?: DataContent;
+    license?: DataContent;
+    language?: DataContent;
+    image?: DataContent;
+    imageType?: DataContent;
+    imageWidth?: DataContent;
+    imageHeight?: DataContent;
+    imageAlt?: DataContent;
+    siteName?: DataContent;
+    institution?: DataContent;
+    twitterSite?: DataContent;
+    twitterCreator?: DataContent;
+}
+
+export interface MetaTagsDefs {
+    [s: string]: DataContent;
+}
+
+export interface NameMetaTagAttrs {
+    name: string; // eslint-disable-line no-restricted-globals
+    content: Content;
+}
+
+export interface PropMetaTagAttrs {
+    property: string;
+    content: Content;
+}
+
+export type MetaTagAttrs = NameMetaTagAttrs | PropMetaTagAttrs;
+
+export interface HeadTagDef {
+    type: string;
+    attrs: MetaTagAttrs;
+}
+
+export default class MetaTags extends Service {
+    @service i18n;
+
+    /**
+     * Get meta tag definitions.
+     *
+     * @method getMetaTags
+     * @param {MetaTagsData} metaTagsOverrides Data values to override defaults.
+     * @return {MetaTagsDefs} Returns meta tag definitions.
+     */
+    getMetaTags(this: MetaTags, metaTagsOverrides: MetaTagsData): MetaTagsDefs {
+        // Default values.
+        const metaTagsData: MetaTagsData = {
+            type: 'collection',
+            description: this.get('i18n').t('general.hosted_on_the_osf'),
+            url: window.location.href,
```

Comment:
Probably something that should be passed in--could be problematic calling window.location in a service, because it's shared